pygen/pygen_src/riscv_utils.py

```diff
@@ -38,18 +35,64 @@ def factory(obj_of):
         sys.exit(1)
 
 
-def get_object(instr):
+def get_object(instr_name):
```

Comment:
would it be possible to to somewhat simplify this function? I think manually checking equality of all the targets and for each individual extension doesn't scale well to when we start adding the 64-bit extensions and so on.

an initial idea I thought of (may not be feasible) would be to first reverse-index the `riscv_instr::instr_group` list to find which instruction group `instr_name` belongs to (e.g. `RV32I`, `RV32C`, so on), then that group name could used used to create the name of the correct instruction package to import.
I think that would save a lot of the effort currently required to manually import every possible package and try creating the instruction object. WDYT?

Reply:
I agree with your proposal Udi, and I think it saves lots of effort (and also that would look better in the code rather than manually checking the equality).

Reply:
Hi @udinator and @hodjat91,
I have thought about this before as well but couldn't able to make it generic with the instr_group. As **`cls.instr_group`** list is updated after the **`create_instr`** function call. So, In order to create an instance for the first instruction, I will get an empty **`cls.instr_group`** list and maybe the corresponding group of the previous instr_name for the successive iteration.

I just figure out another approach, possibly a WA to this particular issue.
During the registration of each instruction, I am thinking to pass two parameters instead of one i.e 
```python3
def register(cls, instr_name, instr_group):
    cls.instr_registry[instr_name] = instr_group
```
In order to register each instruction with the corresponding group as a value like above.

So that I can iterate over the **key** and **value** in the **create_instr_list()** function
```python3
def create_instr_list(cls, cfg):
    for isntr_name, instr_group in cls.instr_registry.items():
        instr_inst = cls.create_instr(instr_name, instr_group)
```

and implement the rest of the thing in the create_instr() function.
This WA will not require manual modification when we add new extensions in the coming days. Like I am doing in **riscv_utils.py** currently.
Let me know if it works or not!

Reply:
Hi @aneels3 

I think your proposed approach should work and is obviously better than fetch things manually (I'm thinking about other alternatives and will let you know if I can come up with a better solution).